src/prepack-cli.js

```diff
@@ -227,16 +227,35 @@ function run(
     let serialized = prepackFileSync(inputFilenames, resolvedOptions);
     processSerializedCode(null, serialized);
   } catch (x) {
-    console.error(x.message);
-    console.error(x.stack);
-    process.exit(1);
+    //FatalErrors must have generated clear CompilerDiagnostic and can safely go to the next codeblock.
+    if (x instanceof FatalError) {
+      invariant(errors.size > 0, "FatalError must generate at least one CompilerDiagnostic");
+    } else {
+      console.error(x.stack);
+      process.exit(1);
+    }
   } finally {
     if (errors.size > 0) {
+      console.error("Errors found while prepacking");
       let foundFatal = false;
       for (let [loc, error] of errors) {
+        let sourceMessage = "";
+        switch (loc.source) {
+          case "":
+            sourceMessage = "In an unknown source file";
+            break;
+          case "no-filename-specified":
+            sourceMessage = "In stdin";
+            break;
+          default:
+            // flow made me do this || ""
+            sourceMessage = `In input file ${loc.source || ""}`;
+            break;
+        }
+
         foundFatal = foundFatal || error.severity === "FatalError";
         console.error(
-          `${loc.source || ""}(${loc.start.line}:${loc.start.column +
+          `${sourceMessage}(${loc.start.line}:${loc.start.column +
```

Comment:
In line 263 below throw a FatalError so that the catch clause (which should now be wrapping the finally), can terminate the process with exit code 1.